src/providers/postgres/qgspostgresprovider.cpp

```diff
@@ -4259,6 +4259,40 @@ static QVariant parseStringArray( const QString &txt )
   return result;
 }
 
+static QVariant parseMultiDimensionnalArray( const QString &txt )
+{
+  QStringList result;
+  if ( !txt.startsWith( '{' ) || !txt.endsWith( '}' ) )
+  {
+    QgsLogger::warning( "Error parsing array, missing curly braces: " + txt );
+    return result;
+  }
+
+  QStringList values;
+  QString text = txt;
+  while ( !text.isEmpty() )
+  {
+    int openedBrackets = 1;
+    int i = 0;
+    while ( i < text.length()  && openedBrackets > 0 )
+    {
+      ++i;
+      if ( text.at( i ) == '}' ) openedBrackets--;
```

Comment:
Ah, what happens if one of the inner string contains a bracket ? does the loop have to ignore escaped brackets ?

Reply:
You're right! commit 5e0dc28 now manage this case